bugsnag-android-core/src/main/java/com/bugsnag/android/FileStore.java

```diff
@@ -138,7 +130,6 @@ String write(@NonNull JsonStream.Streamable streamable) {
 
     void discardOldestFileIfNeeded() {
         // Limit number of saved payloads to prevent disk space issues
-        File storageDir = new File(storeDirectory);
         if (storageDir.isDirectory()) {
             List<File> files = new ArrayList<>(Arrays.asList(storageDir.listFiles()));
```

Comment:
Not a new issue, but I've noticed that `listFiles` can return `null` if there's an IO problem and this would throw an NPE.

Reply:
Updated to add a null check. Also refactored `getFilename()` to avoid requiring `storageDir` to be package private.